apps/openmw/mwworld/worldimp.cpp

```diff
@@ -1024,9 +1025,17 @@ namespace MWWorld
 
             float activationDistance = getMaxActivationDistance() + telekinesisRangeBonus;
 
-            facedObject = getFacedObject(activationDistance);
+            facedObject = getFacedObject(activationDistance, distanceToObject, true);
+
+            // Not allowing telekinesis on actors, on doors that teleport to other cells, or on activators
+            // Original engine doesn't allow telekinesis on books or lights, either
+            if (!facedObject.isEmpty() && (facedObject.getClass().isActor()
+                || facedObject.getCellRef().getTeleport() || facedObject.getClass().getTypeName() == typeid(ESM::Activator).name())
```

Comment:
Perhaps it would be cleanest to move this logic to a new method ```allowTelekinesis``` in MWWorld::Class, implemented in classes like NPC, Activator and Door. That would allow us to get rid of the type check.